modules/collect/src/main/java/com/opengamma/collect/id/LinkResolver.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (C) 2014 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.collect.id;
+
+/**
+ * A resolver that can provide the target when resolving links.
+ * <p>
+ * A {@link Link} provides loose coupling between different parts of the object model.
+ * When the target of a link is needed, it is resolved by passing in a link resolver.
+ * <p>
+ * Link resolution will typically be implemented to access an underlying data store.
+ * If the link specifies an identifier and/or that is not resolvable, an exception is thrown.
```

Comment:
and/or here doesn't make sense. Think something got lost in translation. Original had:

> because the identifier does not exist, or the
> declared type is incorrect